src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs

```diff
@@ -65,33 +66,61 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
                 var nameArg = args[i];
                 var nameArgType = argTypes[i];
 
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
-                // type analysis, so they need to be known upfront (before DropColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String || (nameNode = nameArg.AsStrLit()) == null)
+                // type analysis, so they need to be known upfront (before DropColumns executes).                            
+                if (supportColumnNamesAsIdentifiers)
                 {
-                    fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
-                    continue;
+                    if (nameArg is not FirstNameNode identifierNode)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected an identifier.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedIdentifierArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    expectedColumnName = identifierNode.Ident.Name;
+                }
+                else
+                {
+                    StrLitNode strLitNode = nameArg.AsStrLit();
+
+                    if (nameArgType.Kind != DKind.String && strLitNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    expectedColumnName = strLitNode.Value;
                 }
 
                 // Verify that the name is valid.
-                if (!DName.IsValidDName(nameNode.Value))
+                if (!DName.IsValidDName(expectedColumnName))
                 {
                     fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, nameNode.Value);
+
+                    // Argument '{0}' is not a valid identifier.
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, expectedColumnName);
                     continue;
                 }
 
-                var columnName = new DName(nameNode.Value);
+                var columnName = new DName(nameArg is FirstNameNode && argTypes[0].DisplayNameProvider != null && DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName)
```

Comment:
>argTypes[0].DisplayNameProvider != null [](http://example.com/codeflow?start=71&length=39)

this part is the problem -  DisplayNameProvider  is only 1 of several ways DisplayNames are handled.  For example, this probably would fail on a dataverse table. 

If you're already using the DType API, you shouldn't have to check this.

Reply:
please fix this.